django/db/models/query.py

```diff
@@ -1683,10 +1683,7 @@ def __init__(self, klass_info, select, db):
             reorder_map.sort()
             self.init_list = [v[1] for v in reorder_map]
             pos_list = [row_pos for _, _, row_pos in reorder_map]
```

Comment:
`pos_list` could be changed to be a generator now that it is only unpacked into `itemgetter`.